lib/log.rb

```diff
@@ -80,6 +80,19 @@ def initialize(level=nil)
       else
         @level = INFO
       end
+
+      # Reference variables to evaluate custom error values.
+      @custom_err = ''
+    end
+
+    def write_custom_errmsg(msg)
+      # Capture custom error message during execution. 
+      @custom_err = msg
+    end
+
+    def custom_errmsg
+      # Returns the custom error log in executions.
+      return @custom_err
```

Comment:
```suggestion
attr_accessor :handled_error
```

`attr_accessor` will create automatically reader and writer methods for the specified instance variable. It saves some lines. Just put it somewhere at the top.

My suggestion is to use name `handled_error` to be more clear about the intention of the value.

Reply:
Done